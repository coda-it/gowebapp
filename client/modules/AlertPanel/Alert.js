import React from 'react';
class Alert extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            show: false,
            hide: false,
        };
    }
    componentDidMount() {
        this.showTimeout = setTimeout(() => {
            this.setState({ show: true });
        }, 2000);
        this.hideTimeout = setTimeout(() => {
            this.setState({ hide: true });
        }, 4000);
    }
    componentWillUnmount() {
        if (this.showTimeout) {
            clearTimeout(this.showTimeout);
        }
        if (this.hideTimeout) {
            clearTimeout(this.hideTimeout);
        }
    }
    render() {
        const { type, children } = this.props;
        const { show, hide } = this.state;
        const showClass = show ? 'show' : '';
        const hideClass = hide ? 'hide' : '';
        const classes = `alert-panel__alert alert-panel__alert--${type} ${showClass} ${hideClass}`;
        return React.createElement("div", { className: classes }, children);
    }
}
export default Alert;
//# sourceMappingURL=Alert.js.map