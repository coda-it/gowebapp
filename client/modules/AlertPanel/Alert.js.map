{"version":3,"file":"Alert.js","sourceRoot":"","sources":["Alert.tsx"],"names":[],"mappings":"AAAA,OAAO,KAAK,MAAM,OAAO,CAAC;AAY1B,MAAM,KAAM,SAAQ,KAAK,CAAC,SAAuB;IAC/C,YAAY,KAAK;QACf,KAAK,CAAC,KAAK,CAAC,CAAC;QAEb,IAAI,CAAC,KAAK,GAAG;YACX,IAAI,EAAE,KAAK;YACX,IAAI,EAAE,KAAK;SACZ,CAAC;IACJ,CAAC;IAED,iBAAiB;QAEf,IAAI,CAAC,WAAW,GAAG,UAAU,CAAC,GAAG,EAAE;YACjC,IAAI,CAAC,QAAQ,CAAC,EAAE,IAAI,EAAE,IAAI,EAAE,CAAC,CAAC;QAChC,CAAC,EAAE,IAAI,CAAC,CAAC;QAET,IAAI,CAAC,WAAW,GAAG,UAAU,CAAC,GAAG,EAAE;YACjC,IAAI,CAAC,QAAQ,CAAC,EAAE,IAAI,EAAE,IAAI,EAAE,CAAC,CAAC;QAChC,CAAC,EAAE,IAAI,CAAC,CAAC;IACX,CAAC;IAED,oBAAoB;QAElB,IAAI,IAAI,CAAC,WAAW,EAAE;YAEpB,YAAY,CAAC,IAAI,CAAC,WAAW,CAAC,CAAC;SAChC;QAED,IAAI,IAAI,CAAC,WAAW,EAAE;YAEpB,YAAY,CAAC,IAAI,CAAC,WAAW,CAAC,CAAC;SAChC;IACH,CAAC;IAED,MAAM;QACJ,MAAM,EAAE,IAAI,EAAE,QAAQ,EAAE,GAAG,IAAI,CAAC,KAAK,CAAC;QACtC,MAAM,EAAE,IAAI,EAAE,IAAI,EAAE,GAAG,IAAI,CAAC,KAAK,CAAC;QAElC,MAAM,SAAS,GAAG,IAAI,CAAC,CAAC,CAAC,MAAM,CAAC,CAAC,CAAC,EAAE,CAAC;QACrC,MAAM,SAAS,GAAG,IAAI,CAAC,CAAC,CAAC,MAAM,CAAC,CAAC,CAAC,EAAE,CAAC;QACrC,MAAM,OAAO,GAAG,0CAA0C,IAAI,IAAI,SAAS,IAAI,SAAS,EAAE,CAAC;QAE3F,OAAO,6BAAK,SAAS,EAAE,OAAO,IAAG,QAAQ,CAAO,CAAC;IACnD,CAAC;CACF;AAED,eAAe,KAAK,CAAC"}