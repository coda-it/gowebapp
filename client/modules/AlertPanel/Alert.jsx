// @flow
import React from 'react';

type Props = {
  type: string,
  children: React$Element<any>,
};

type State = {
  show: boolean,
  hide: boolean,
};

class Alert extends React.Component<Props, State> {
  constructor() {
    super();

    this.state = {
      show: false,
      hide: false,
    };
  }

  componentDidMount() {
    // $FlowFixMe - this component will be transformed into typescript soon
    this.showTimeout = setTimeout(() => {
      this.setState({ show: true });
    }, 2000);
    // $FlowFixMe - this component will be transformed into typescript soon
    this.hideTimeout = setTimeout(() => {
      this.setState({ hide: true });
    }, 4000);
  }

  componentWillUnmount() {
    // $FlowFixMe - this component will be transformed into typescript soon
    if (this.showTimeout) {
      // $FlowFixMe - this component will be transformed into typescript soon
      clearTimeout(this.showTimeout);
    }
    // $FlowFixMe - this component will be transformed into typescript soon
    if (this.hideTimeout) {
      // $FlowFixMe - this component will be transformed into typescript soon
      clearTimeout(this.hideTimeout);
    }
  }

  render() {
    const { type, children } = this.props;
    const { show, hide } = this.state;

    const showClass = show ? 'show' : '';
    const hideClass = hide ? 'hide' : '';
    const classes = `alert-panel__alert alert-panel__alert--${type} ${showClass} ${hideClass}`;

    return <div className={classes}>{children}</div>;
  }
}

export default Alert;
