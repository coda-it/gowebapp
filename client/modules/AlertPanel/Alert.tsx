import React from 'react';

type Props = {
  type: string;
  children: React.ReactNode;
};

type State = {
  show: boolean;
  hide: boolean;
};

class Alert extends React.Component<Props, State> {
  constructor(props) {
    super(props);

    this.state = {
      show: false,
      hide: false,
    };
  }

  componentDidMount() {
    // @ts-ignore
    this.showTimeout = setTimeout(() => {
      this.setState({ show: true });
    }, 2000);
    // @ts-ignore
    this.hideTimeout = setTimeout(() => {
      this.setState({ hide: true });
    }, 4000);
  }

  componentWillUnmount() {
    // @ts-ignore
    if (this.showTimeout) {
      // @ts-ignore
      clearTimeout(this.showTimeout);
    }
    // @ts-ignore
    if (this.hideTimeout) {
      // @ts-ignore
      clearTimeout(this.hideTimeout);
    }
  }

  render() {
    const { type, children } = this.props;
    const { show, hide } = this.state;

    const showClass = show ? 'show' : '';
    const hideClass = hide ? 'hide' : '';
    const classes = `alert-panel__alert alert-panel__alert--${type} ${showClass} ${hideClass}`;

    return <div className={classes}>{children}</div>;
  }
}

export default Alert;
