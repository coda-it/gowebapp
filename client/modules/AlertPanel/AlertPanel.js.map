{"version":3,"file":"AlertPanel.js","sourceRoot":"","sources":["AlertPanel.tsx"],"names":[],"mappings":"AAAA,OAAO,CAAC,MAAM,QAAQ,CAAC;AACvB,OAAO,KAAK,MAAM,OAAO,CAAC;AAE1B,OAAO,KAAK,MAAM,SAAS,CAAC;AAM5B,MAAM,UAAU,GAAG,CAAC,KAAY,EAAE,EAAE;IAClC,MAAM,EAAE,MAAM,EAAE,GAAG,KAAK,CAAC;IAEzB,OAAO,CACL,6BAAK,SAAS,EAAC,aAAa,IACzB,CAAC,CAAC,GAAG,CAAC,MAAM,EAAE,CAAC,KAAK,EAAE,KAAK,EAAE,EAAE;QAC9B,MAAM,EAAE,IAAI,EAAE,OAAO,EAAE,GAAG,KAAK,CAAC;QAChC,MAAM,GAAG,GAAG,SAAS,KAAK,EAAE,CAAC;QAE7B,OAAO,CACL,oBAAC,KAAK,IAAC,GAAG,EAAE,GAAG,EAAE,IAAI,EAAE,IAAI,IACxB,OAAO,CACF,CACT,CAAC;IACJ,CAAC,CAAC,CACE,CACP,CAAC;AACJ,CAAC,CAAC;AAEF,eAAe,UAAU,CAAC"}