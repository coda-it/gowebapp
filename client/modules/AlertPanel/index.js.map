{"version":3,"file":"index.js","sourceRoot":"","sources":["index.ts"],"names":[],"mappings":"AAAA,OAAO,EAAE,OAAO,EAAE,MAAM,aAAa,CAAC;AACtC,OAAO,KAAK,SAAS,MAAM,gCAAgC,CAAC;AAC5D,OAAO,UAAU,MAAM,cAAc,CAAC;AAEtC,MAAM,eAAe,GAAG,CAAC,KAAK,EAAE,EAAE;IAChC,MAAM,MAAM,GAAG,SAAS,CAAC,gBAAgB,CAAC,KAAK,CAAC,CAAC;IAEjD,OAAO;QACL,MAAM;KACP,CAAC;AACJ,CAAC,CAAC;AAEF,eAAe,OAAO,CAAC,eAAe,EAAE,IAAI,CAAC,CAAC,UAAU,CAAC,CAAC"}