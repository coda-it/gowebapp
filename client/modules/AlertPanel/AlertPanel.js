import _ from 'lodash';
import React from 'react';
import Alert from './Alert';
const AlertPanel = (props) => {
    const { alerts } = props;
    return (React.createElement("div", { className: "alert-panel" }, _.map(alerts, (alert, index) => {
        const { type, message } = alert;
        const key = `alert-${index}`;
        return (React.createElement(Alert, { key: key, type: type }, message));
    })));
};
export default AlertPanel;
//# sourceMappingURL=AlertPanel.js.map