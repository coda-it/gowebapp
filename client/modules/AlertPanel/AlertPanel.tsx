import React from 'react';
import { useSelector } from 'react-redux';
import * as selectors from 'client/models/alerts/selectors';
import Alert from './Alert';

const AlertPanel = () => {
  const alerts = useSelector(selectors.getLimitedAlerts);

  return (
    <div className="alert-panel">
      {alerts.map((alert, index) => {
        const { type, message } = alert;
        const key = `alert-${index}`;

        return (
          <Alert key={key} type={type}>
            {message}
          </Alert>
        );
      })}
    </div>
  );
};

export default AlertPanel;
