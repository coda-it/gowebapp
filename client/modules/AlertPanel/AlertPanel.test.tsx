import React from 'react';
import { mount } from 'enzyme';
import * as constants from 'client/models/alerts/constants';
import AlertPanel from './AlertPanel';

describe('AlertPanel/AlertPanel', () => {
  it('should render correctly', () => {
    const alerts = [
      {
        type: 'info',
        message: 'Message 1',
        timestamp: new Date('2020-03-01'),
        isOld: false,
      },
      {
        type: 'info',
        message: 'Message 2',
        timestamp: new Date('2020-03-01'),
        isOld: false,
      },
      {
        type: 'info',
        message: 'Message 3',
        timestamp: new Date('2020-03-01'),
        isOld: false,
      },
    ];

    const component = mount(<AlertPanel alerts={alerts} />);

    expect(component).toMatchSnapshot();
  });
});
