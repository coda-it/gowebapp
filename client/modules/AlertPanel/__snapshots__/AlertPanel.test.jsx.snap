// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertPanel/AlertPanel should render correctly 1`] = `
<div
  class="alert-panel"
/>
`;
